src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -488,7 +490,7 @@ public StructImpl download(
               canonicalId,
               Optional.<String>absent(),
               outputPath.getPath(),
-              env.getListener(),
+              allowFail ? NullEventHandler.INSTANCE : env.getListener(),
```

Comment:
1. could you make the same change for `download_and_extract`?
2. could you add a short note in the documentation of `allow_fail` explaining that it silences the warnings as well?

Reply:
Done.